code/datums/beam.dm

```diff
@@ -88,7 +94,7 @@
 	return ..()
 
 /datum/beam/proc/Draw()
-	var/Angle = round(Get_Angle(get_turf(origin), get_turf(target)))
+	var/Angle = round(Get_Angle(origin.x ? origin : get_turf(origin), target.x ? target : get_turf(target)))
```

Comment:
Could `origin` or `target` possibly still be `null` here (perhaps when it's not set in `new`?

I'd consider this just to be sure:
```suggestion
	var/Angle = round(Get_Angle(origin?.x ? origin : get_turf(origin), target?.x ? target : get_turf(target)))
```

Reply:
If it *is* `null` then something has been called improperly and we want the runtime to happen. `target` and `origin` are set in `/datum/beam/New()`

Also if we do `target?.x ? target : get_turf(target)` then the ternary might try to call `get_turf(target)` when `target` is null... which is going to have the same result (runtime).

Reply:
Fair enough. 👍